utils/wheel_feeder.py

```diff
@@ -228,44 +361,88 @@ def main():
     print("Initializing the Wheel Feeder...")
     parser = argparse.ArgumentParser(
         description=DESCRIPTION,
-        formatter_class=argparse.RawTextHelpFormatter
+        formatter_class=argparse.RawTextHelpFormatter,
+        epilog="""
+
+%(program)s -r <region> -s <stack name> -n <wheel name> -c <csv file path>
+
+%(program)s -r <region> -u <wheel url> -w <wheel uuid> -p <cognito group id> -i <cognito client id> -c <csv file path>
+""" % {"program": sys.argv[0]}
     )
 
     parser.add_argument(
-        '-u', '--wheel-url', required=True,
+        '-u', '--wheel-url', required=False,
         help='Full URL of the Wheel\'s API Gateway endpoint. \n'
         'Example: https://<API_ID>.execute-api.us-west-2.amazonaws.com'
     )
     parser.add_argument(
-        '-w', '--wheel-id', required=True,
+        '-w', '--wheel-id', required=False,
         help='UUID of the Wheel which you want to feed. \n'
         'Example: 57709419-17c9-4b77-ac99-77fb0d7c7c51'
     )
     parser.add_argument(
-        '-c', '--csv-file-path', required=True,
+        '-c', '--csv-file-path', required=False,
         help='Path to the CSV file. \n'
         'Example: /home/foo/participants.csv'
     )
     parser.add_argument(
-        '-p', '--cognito-user-pool-id', required=True,
+        '-p', '--cognito-user-pool-id', required=False,
         help='Cognito User Pool Id. \n'
         'Example: us-west-2_K4oiNOTREAL'
     )
     parser.add_argument(
-        '-i', '--cognito-client-id', required=True,
+        '-i', '--cognito-client-id', required=False,
         help='Cognito Client Id (get it by visiting your Cognito User Pool). \n'
         'Example: 6e6p1k4qaNOTREAL'
     )
+    parser.add_argument(
+        '-s', '--stack-name', required=False,
+        help='The CloudFormation stack name used to set up your wheel.\n'
+        'Example: MyAWSWheel Stack'
+    )
+    parser.add_argument(
+        '-r', '--region', required=False, default="us-west-2",
+        help='The AWS Region in which your Wheel stack lives, defaults to us-west-2\n'
+        'Example: us-west-2'
+    )
+    parser.add_argument(
+        '-n', '--wheel-name', required=False,
+        help='The name of your Wheel\n'
+        'Example: MyOrganizationWheel'
+    )
     args = parser.parse_args()
 
-    # Initialize the Feeder and execute it.
-    wheel_feeder = WheelFeeder(
-        args.wheel_url,
-        args.wheel_id,
-        args.csv_file_path,
-        args.cognito_user_pool_id,
-        args.cognito_client_id
-    )
+    if args.wheel_name and args.stack_name and args.region:
+        metadata = WheelMetadataProvider(args.stack_name,
+                                         args.region,
+                                         args.wheel_name)
+        print("url=%s id=%s userpoolid=[%s] clientid=[%s]" % (metadata.wheel_url,
```

Comment:
You have a lot of verbose code you don't really require because it's in the 'Outputs' of the cloud formation stack, so let's simplify this a lot.  We'll need to make sure that WheelFeeder knows it doesn't need the STAGE_IDENTIFIER from the Wheel feeder as a part of this, but it'll make it even easier to follow and we can construct the wheel feeder the same way in both cases.  This is just filling in some defaults if you specify the stack name.

```python
if args.stack_name:
    cf_client = boto3.client('cloudformation', region_name=args.region).
    stack = cf_client.describe_stacks(StackName=args.stack_name)['Stacks'][0]
    stack_outputs = {output['OutputKey']: output['OutputValue'] for output in stack['Outputs']}
    args.cognito_client_id = args.cognito_client_id or stack_outputs['CognitoUserPoolClient']
    args.cognito_user_pool_id = args.cognito_user_pool_id or stack_outputs['CognitoUserPool']
    if args.wheel_url:
        full_wheel_url = args.wheel_url.rstrip('/') + '/app'
    else:
        full_wheel_url = stack_outputs['Endpoint']
    if args.wheel_name:
        ddb_client = boto3.resource('dynamodb', region_name=args.region)
        for item in ddb_client.Table(stack_outputs['wheelDynamoDBTable']).scan()['Items']:
            if item['name'] == args.wheel_name:
                args.wheel_id = item['id']
                break
        else:
            raise SystemExit("ERROR: Could not find a wheel with the name '%s'" % args.wheel_name")
```